src/arch/linux/mapping/mapping.c

```diff
@@ -860,7 +863,8 @@ int register_hardware_ram(int type, unsigned base, unsigned int size)
 void register_hardware_ram_virtual2(int type, unsigned base, unsigned int size,
 	void *uaddr, dosaddr_t va)
 {
-  do_register_hwram(type, base, size, MEM_BASE32(va), va);
+  do_register_hwram(type, base, size,
```

Comment:
So why still not _virtual2()?

Reply:
I am not sure I understand? MEM_BASE32x is a static function, so using _virtual() instead of _virtual2() to map kvm_base.